infra/main.bicep

```diff
@@ -0,0 +1,117 @@
+targetScope = 'subscription'
+
+@minLength(1)
+@maxLength(64)
+@description('Name of the the environment which is used to generate a short unique hash used in all resources.')
+param environmentName string
+
+@minLength(1)
+@description('Primary location for all resources (filtered on available regions for Azure Open AI Service).')
+@allowed([ 'westeurope', 'southcentralus', 'australiaeast', 'canadaeast', 'eastus', 'eastus2', 'francecentral', 'japaneast', 'northcentralus', 'swedencentral', 'switzerlandnorth', 'uksouth' ])
```

Comment:
/rant 
We (not you Chris on your own) _really_ need to come up with a better way than hardcoding the list.
Or at a minimum, we need to monitor the available regions along with the repos that have hardcoded lists and update the repos whenever the list changes. Not awesome!